src/RedisCache/RedisCache/Commands/SetAzureRedisCache.cs

```diff
@@ -36,7 +36,7 @@ public class SetAzureRedisCache : RedisCacheCmdletBase
         public string Name { get; set; }
 
         [Parameter(ValueFromPipelineByPropertyName = true, Mandatory = false, HelpMessage = "Size of redis cache. Valid values: P1,P2, P3, P4, P5, C0, C1, C2, C3, C4, C5, C6, 250MB, 1GB, 2.5GB, 6GB, 13GB, 26GB, 53GB, 120GB")]
-        [PSArgumentCompleter(SizeConverter.P1String, SizeConverter.P2String, SizeConverter.P3String, SizeConverter.P4String, SizeConverter.P5String,
+        [PSArgumentCompleter, ValidateSet(SizeConverter.P1String, SizeConverter.P2String, SizeConverter.P3String, SizeConverter.P4String, SizeConverter.P5String,
```

Comment:
To fix the issue: https://github.com/Azure/azure-powershell/issues/9510. We need to validate the Size param. Argument Completer isn't validating the param and allows unsupported values, so ValidateSet has been added.

Regarding analysis failure, I have already submitted breaking change issue and it is closed now: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/428. Please let me know what else needs to be done for fixing the analysis failure.

Reply:
> To fix the issue #9510, we need to validate the Size param. Argument Completer isn't validating the param and allows unsupported values, so ValidateSet has been added.
> 
> Regarding analysis failure, I have already submitted breaking change issue and it is closed now: [Azure/azure-powershell-cmdlet-review-pr#428](https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/428). Please let me know what else needs to be done for fixing the analysis failure.

The breaking change has a long process to go GA. Firstly you need to add breaking change notice to master and then create a preview branch to contain your breaking change. We have breaking change releases twice a year. One just passed and the next will be 6 months later. For your case, since you have already use a argument completer, user is unlikely to input an invalid value. Further, the invalid input value is not the case related to the issue you refer to.

Reply:
Fixed the breaking change issues